plugins/request.rb

```diff
@@ -2,23 +2,72 @@
 
 module Msf
 
-class Plugin::HTTPRequests < Msf::Plugin
+class Plugin::Requests < Msf::Plugin
 
   class ConsoleCommandDispatcher
     include Msf::Ui::Console::CommandDispatcher
 
     def name
-      'HTTP Requests'
+      'Request'
     end
 
     def commands
       {
-        'httpr' => 'Make an HTTP request'
+        'request' => "Make a request of the specified type (#{types.join(', ')})",
       }
     end
 
-    def parse_args(args)
-      help_line = 'Usage: httpr [options] uri'
+    def types
+      # dynamically figure out what types are supported based on parse_args_*
+      parse_methods = self.public_methods.select {|m| m.to_s =~ /^parse_args_/}
+      parse_methods.collect {|m| m.to_s.split('_').slice(2..-1).join('_')}
+    end
+
+    def cmd_request(*args)
+      # grab and validate the first arg as type, which will affect how the
+      # remaining args are parsed
+      type = args.shift
+      # short circuit the whole deal if they need help
+      return help if (!type || type =~ /^-?-h(?:elp)?$/)
+      type.downcase!
+      opts, opt_parser = parse_args(type, args)
+
+      if opts && opt_parser
+        # handle any "global" options
+        if opts[:output_file]
+          begin
+            opts[:output_file] = File.new(opts[:output_file], 'w')
+          rescue ::Errno::EACCES, Errno::EISDIR, Errno::ENOTDIR
+            return help(opt_parser, 'Failed to open the specified file for output')
+          end
+        end
+        # hand off the actual request to the appropriate request handler
+        handler_method = "handle_request_#{type}".to_sym
+        if self.respond_to?(handler_method)
+          # call the appropriate request handler
+          self.send(handler_method, opts, opt_parser)
+        else
+          help(opt_parser, "No request handler found for type (#{type.to_s}).")
+        end
+      else
+        help(opt_parser, "No valid options provided for request #{type}")
+      end
+    end
+
+    def parse_args(type, args)
+      type.downcase!
```

Comment:
this was more of a guard just in case someone calls parse_args independently, like after us.  I agree it's currently redundant, but thought it might be safer in the future this way.  I certainly don't care one way or the other however, so I leave it up to you